### PR TITLE
Added implementation of HTMLCollection for Element.getElementsByTagName

### PR DESCRIPTION
**Pull request by Indomian**
- Added class HTMLCollection with implementation of item() method.
 - Added corresponded tests.

Signed-off-by: Egor Bolgov <e.bolgov@semrush.com>

### CODE REVIEW

File: src/utils/HTMLCollection.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+export default class HTMLCollection extends Array {
+    constructor(parent, selector) {
+        super();
+        this._parent = parent;
+        this._selector = selector;
+        this._fillChildren(this._parent);
+    }
+
+    _fillChildren(parent) {
```
Comment (christophehurpeau) [inline src/utils/HTMLCollection.js]: HtmlCollection is supposed to be a live collection: I suppose that this implementation doesn't work

test:

```js
const document = new Document();
const divCollection = document.body.getElementsByTagName('div');
document.body.appendChild(document.createElement('div'));
strictEqual(divCollection.length, 1);
```

Comment (Indomian) [inline src/utils/HTMLCollection.js]: Yep, I know that. This kind of next step. I required HTMLCollection as interface only (especially `item()` method). Hope later I will make it live one.

Comment (christophehurpeau) [inline src/utils/HTMLCollection.js]: I think it would be better to implement it directly as the specification

Comment (Indomian) [inline src/utils/HTMLCollection.js]: Updated brunch - looks like ok for simple tests.

Comment (Indomian) [inline src/utils/HTMLCollection.js]: Funny :) While I decided to make live version you suggested to do it :)

File: src/ParentNode.js [hunk 2]
```diff
@@ -308,6 +338,7 @@ export default class ParentNode extends Node {
      */
     set innerHTML(html) {
         this._childNodes = [];
+        this._updateCollections();
```
Comment (christophehurpeau) [inline src/ParentNode.js]: maybe should be after parse ?

Comment (Indomian) [inline src/ParentNode.js]: Nope, parse itself calls appendChild, etc. and structure updates automatically. I think we should clear all collections before parse to unlink elements that left in collections but already removed from DOM tree.

**Comment by Indomian** [comment]
@christophehurpeau Any chance of merging this?

**Comment by Indomian** [comment]
Thank you!

**Pull request merged**